### PR TITLE
Fix error message for DOLT_MERGE conflicts

### PR DESCRIPTION
**Pull request by VinaiRachakonda**
This pr is a little incomplete with regards to constrain violations. cc: https://github.com/dolthub/dolt/issues/3414

### CODE REVIEW

File: go/libraries/doltcore/doltdb/errors.go [hunk 1]
```diff
@@ -46,8 +46,10 @@ var ErrUpToDate = errors.New("up to date")
 var ErrIsAhead = errors.New("current fast forward from a to b. a is ahead of b already")
 var ErrIsBehind = errors.New("cannot reverse from b to a. b is a is behind a already")
 
+var ErrUnresolvedConflictsAutocommit = errors.New("merge conflict with autocommit on. Aborting merge. To resolve conflicts with dolt_merge, do so in a single transaction. To SQL commit conflicts set @@dolt_allow_commit_conflicts=1")
```
[the remark this answers is not in the document]
Comment (zachmu) [reply go/libraries/doltcore/doltdb/errors.go:49]: Move this and the other new error into the SQL package, they are specific to how we handle merge conflicts in the SQL context

File: go/libraries/doltcore/sqle/dfunctions/dolt_merge.go [hunk 2]
```diff
@@ -175,7 +180,14 @@ func mergeIntoWorkingSet(ctx *sql.Context, sess *dsess.DoltSession, roots doltdb
 					return ws, hasConflicts, wsErr
 				}
 
-				ctx.Warn(DoltMergeWarningCode, err.Error())
+				// Give a better error message for unresolved conflicts when autocommit is on. This is because the transaction
+				// gets aborted when merge conflicts arise during autocommit and the dolt_conflicts table is not actually
+				// written to.
+				if autocommit {
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/dfunctions/dolt_merge.go:186]: Don't write different error messages for autocommit or not. Giving more details error messages is fine.

**Review by zachmu — COMMENTED** [review]
This isn't the right place to check for this error or anything about it. That logic lives in the transaction code.

Also don't want to have logic that does different things for autocommit = on or off. There's nothing magic or special about autocommit. All these errors can happen with autocommit off too.

**Comment by timsehn** [comment]
In @VinaiRachakonda's defense, I told him to have a different error message for autocommit on/off since many users have hit this and been confused. 

@zachmu 's proposed error message is fine to try. But if users still are confused after seeing it, I'm overruling him.

**Review by zachmu — APPROVED** [review]
Just a couple new copy edits, otherwise LGTM

**Pull request merged**